Add note about Vaadin scoped beans and serialization (#4244)
vaadin/docs: articles/flow/integrations/cdi/contexts.adoc

The `UI` instance itself isn't preserved, but routing components are preserved. Any bean tied to the `UI` instance with `UIScoped` is recreated. The preserved routing components are moved to the new `UI`. To preserve beans during a browser refresh, you need to use `@RouteScoped`, as shown earlier.

.Vaadin scoped bean need to be serializable
[NOTE]

> The [classname]`BeanStore` is stored in the session. This means in case you persist sessions using some 3rd party tool, the beans needs to be serializable.

Reviewer: There is no `BeanStore` in CDI. Maybe just talk about bean instances?
```suggestion
Scoped bean instances are stored in the session. This means in case you persist sessions using some 3rd party tool, the beans needs to be serializable.
```
